browser/extensions/api/brave_rewards_api.cc

```diff
@@ -133,7 +133,10 @@ ExtensionFunction::ResponseAction BraveRewardsGetGrantFunction::Run() {
   RewardsService* rewards_service_ =
     RewardsServiceFactory::GetForProfile(profile);
   if (rewards_service_) {
-    rewards_service_->FetchGrant(std::string(), std::string());
+    bool should_prevent = rewards_service_->GetPreventGrant();
```

Comment:
this doesn't seem like logic that should be in the UI. The status of the claimed grant should be tracked internally in ledger and it should respond with an appropriate error for the FetchGrant response